### PR TITLE
Correct code snippet in README

### PR DESCRIPTION
**Pull request by Snuggle**
Hello!

I noticed that after copy/pasting the GitHub action snippet, this `if` check does not work.

I've replaced it with `if: failure()` which seems to actually work now. Please feel free to make any edits needed to this pull request!

### CODE REVIEW

**Comment by mre** [comment]
Weird, I thought we fixed that for good in https://github.com/lycheeverse/lychee-action/pull/102. 🤔
`failure()` sounds more like it unconditionally triggers a failure, but I haven't looked it up yet.

**Comment by polarathene** [comment]
**TL;DR:**
- This is semi-correct, especially considering the [2.0 plan to flip the default for the `fail` input](https://github.com/lycheeverse/lychee-action/pull/85) (_that input doesn't seem worth keeping as described below_).
- An improvement would be to add an extra condition to check for the `lychee-action` step status specifically, just in-case the workflow implemented by the user has earlier steps that could fail the job too (_see below example_).

---

> `failure()` sounds more like it unconditionally triggers a failure, but I haven't looked it up yet.

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-status-check-functions
https://docs.github.com/en/actions/learn-github-actions/expressions#failure

> Returns `true` when any previous step of a job fails. If you have a chain of dependent jobs, `failure()` returns `true` if any ancestor job fails.
> 
> **failure with conditions**
> You can include extra conditions for a step to run after a failure,
> but you must still include `failure()` to override the default status check of `success()` that is automatically applied to `if` conditions that don't contain a status check function.

The linked docs have an example for "failure with conditions" that shows that you can check if a specific step failed:

```yaml
if: ${{ failure() && steps.lychee-action.conclusion == 'failure' }}
```

---

Alternatively one can use `continue-on-error` ([`step` context docs](https://docs.github.com/en/actions/learn-github-actions/contexts#steps-context)):

> When a `continue-on-error` step fails, the `outcome` is `failure`, but the final `conclusion` is `success`.

[`step.continue-on-error`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error):

> - Prevents a job from failing when a step fails.
> - Set to `true` to allow a job to pass when this step fails.

Which AFAIK could be used with the `lychee-action` step, inversely providing the same [purpose of the `fail` input for this action](https://github.com/lycheeverse/lychee-action/pull/64/files)? (and aligned instead of inverse, with [the planned 2.0 change for `fail` input](https://github.com/lycheeverse/lychee-action/pull/85)):

> - Fail action on broken links
> - Fail entire pipeline on error (i.e. when lychee exit code is not 0)

https://github.com/lycheeverse/lychee-action/blob/76ab977fedbeaeb32029313724a2e56a8a393548/entrypoint.sh#L44-L51

That is... by default always exiting the action with the status code from the `lychee` command (_as if `lychee-action` had `fail: true` input as default, like it will in 2.0_).

To allow failure for the step and continue the job, users opt-in by adding `continue-on-error: true` instead, which is more descriptive of what to expect and consistent with other actions?

Likewise, instead of `continue-on-error: true` on the `lychee-action` step, users can define later steps with the `failure()` conditional to only run those steps in response to a prior step failing (_which as shown above can specifically reference status via the `conclusion` of a step_).

---

**`continue-on-error` vs `failure()`:**
- `step-a.continue-on-error: true` => `step-a.outcome == failure` && `step-a.conclusion == success`. `step-b` will run even if `step-a` failed.
- `step-a.outcome == failure` is valid for `step-b.if: failure()`, while `step-a.conclusion` may differ. `step-b` will only run if `step-a` failed (_useful for conditionally reporting an issue_).

File: README.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ jobs:
           GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
 
       - name: Create Issue From File
-        if: steps.lychee.outputs.exit_code != 0
+        if: failure()
```
Comment (polarathene) [inline README.md:41]: Either defer this PR change to 2.0, or also update `lychee` step to set `with: fail: true`.

```suggestion
        if: failure() && steps.lychee.outcome == 'failure'
```

**NOTE:** This example currently should only work properly if `steps.lychee.with.fail: true` (_the prior `exit_code` output did not require that_), until 2.0 release makes that default, or more correctly drops `fail` input and defers to `steps.lychee.continue-on-error: true` for the 2.0 `fail: false` behaviour.

**Review by polarathene — CHANGES_REQUESTED** [review]

**Comment by polarathene** [comment]
> this `if` check does not work.

I have since [realized that the `exit_code` output is internal to `lychee-action`](https://github.com/lycheeverse/lychee-action/pull/131#issuecomment-1173409165), it's not actually making it available to other steps.

---

I'm also on the fence about `failure()` being appropriate, as the intent here is for opening an issue when `exit_code == 2`, not other failures which you might want to be notified about (_GH at least emails me when a workflow job fails AFAIK, and will visualize it as a failure_).

In this scenario, the outcome you want is to create an issue when `lychee-action` finds link errors, if you don't have any other outcomes to support (_eg: that is the sole purpose of the job and it's running on a schedule_), then it should not be logged/reported as a failure to the CI (_the issue creation is handling that responsibility_), but successful as the failure is expected to be found and delegated/reported, while still being successful if no errors are found.

Whereas with a lint task on a PR activity, you're interested in the check-suite status. A failure for a job there is more meaningful,  and a simple way of showing what tasks failed.

**Comment by mre** [comment]
If someone finds the time, I'd be thankful for a review of #145, which is another attempt to fix this issue.

**Comment by polarathene** [comment]
I will review later today. I should be able to spare some more time later this week to make some contributions too hopefully.

**Comment by mre** [comment]
#145 has landed, which should hopefully fix that issue. It was apparently broken since we moved to a composite action. @Snuggle can you give `master` another try?

**Comment by mre** [comment]
Gonna close this as fixed. Thanks to you both for your contributions. 😊 Please don't hesitate to open more issues or pull requests.
